src/main/java/org/kiwiproject/registry/eureka/common/EurekaInstance.java

```diff
@@ -129,6 +131,9 @@ public ServiceInstance toServiceInstance() {
                 buildApplicationPortOrNull(port, Security.NOT_SECURE),
                 buildApplicationPortOrNull(securePort, Security.SECURE));
 
+        var upSince = nonNull(leaseInfo) && leaseInfo.containsKey("serviceUpTimestamp")
```

Comment:
Shouldn't we _require_ this?

Reply:
I just looked and for some reason the `.json` sample files in `src/test/resources/EurekaParserTest` don't have all the expected `leaseInfo` elements...

Reply:
I'm not sure if Eureka will always send that map or not.  Remember this gets loaded by Jackson.

Reply:
OK, I just looked at `EurekaParser#buildInstance` and it always sets `leaseInfo`  (see `extractLeaseInfo`). So if we assume a `EurekaInstance` always goes through there, then we would always have `leaseInfo` and a `serviceUpTimestamp`. But if (for whatever reason) the response from Eureka does not contain `serviceUpTimestamp` we default it to zero...which maybe we could also do the same for Consul.

Reply:
Eureka will return 0 for the uptime if the service does not have a status of UP

Reply:
Or maybe we _don't_ assume it always goes through `EurekaParser` and instead we just default the `serviceUpTimestamp` to zero if we don't find it. Yay / nay?

e.g.

```java
        var upSince = nonNull(leaseInfo) && leaseInfo.containsKey("serviceUpTimestamp") 
                ? Instant.ofEpochMilli((long) leaseInfo.get("serviceUpTimestamp")) : Instant.EPOCH;
```